polkadot/node/network/approval-distribution/src/lib.rs

```diff
@@ -2446,46 +2512,80 @@ impl ApprovalDistribution {
 							return
 						},
 					};
-					match message {
-						FromOrchestra::Communication { msg } =>
-							Self::handle_incoming(&mut ctx, state, msg, &self.metrics, rng).await,
-						FromOrchestra::Signal(OverseerSignal::ActiveLeaves(update)) => {
-							gum::trace!(target: LOG_TARGET, "active leaves signal (ignored)");
-							// the relay chain blocks relevant to the approval subsystems
-							// are those that are available, but not finalized yet
-							// activated and deactivated heads hence are irrelevant to this subsystem, other than
-							// for tracing purposes.
-							if let Some(activated) = update.activated {
-								let head = activated.hash;
-								let approval_distribution_span =
-									jaeger::PerLeafSpan::new(activated.span, "approval-distribution");
-								state.spans.insert(head, approval_distribution_span);
-							}
-						},
-						FromOrchestra::Signal(OverseerSignal::BlockFinalized(_hash, number)) => {
-							gum::trace!(target: LOG_TARGET, number = %number, "finalized signal");
-							state.handle_block_finalized(&mut ctx, &self.metrics, number).await;
-						},
-						FromOrchestra::Signal(OverseerSignal::Conclude) => return,
-					}
+
+
+					self.handle_from_orchestra(message, &mut approval_voting_sender, &mut network_sender, state, rng).await;
+
 				},
 			}
 		}
 	}
 
-	async fn handle_incoming<Context>(
-		ctx: &mut Context,
+	/// Handles a from orchestra message received by approval distribution subystem.
+	pub async fn handle_from_orchestra<
+		N: overseer::SubsystemSender<NetworkBridgeTxMessage>,
+		A: overseer::SubsystemSender<ApprovalVotingMessage>,
```

Comment:
This won't work because in follow-up PRs this channels would be different, the approval-voting one would be pointing to the worker, since there won't be any approval-voting subsystem.